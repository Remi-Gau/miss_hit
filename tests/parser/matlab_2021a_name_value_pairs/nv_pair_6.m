% (c) Copyright 2021 Florian Schanda

a = 42;
potato(a=5);
disp(a);

% Ahhhhh.
% * Octave prints '5' and '5'
% * MATLAB < 2021a errors out "The expression to the left of the
%   equals sign is not a valid target for an assignment."
% * MATLAB 2021a prints 'a', and '5', and '42'

% This is so comically bad, I don't even know where to start. I *did*
% warn MathWorks about this, but clearly they don't care.
