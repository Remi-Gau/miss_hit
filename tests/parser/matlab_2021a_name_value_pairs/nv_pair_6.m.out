----------------------------------------------------------------------
--  Parse tree for nv_pair_6.m
Root: Script_File
  Statements: Sequence_Of_Statements
    Statements: Simple_Assignment_Statement
      LHS: Identifier <a>
      RHS: Number_Literal <42>
    Statements: Naked_Expression_Statement
      Expression: Reference
        Name: Identifier <potato>
        Arguments: Name_Value_Pair
          Name: Identifier <a>
          Value: Number_Literal <5>
    Statements: Naked_Expression_Statement
      Expression: Reference
        Name: Identifier <disp>
        Arguments: Identifier <a>
----------------------------------------------------------------------
MISS_HIT Debug Summary: 1 file(s) analysed, everything seems fine
